**VERDICT: PASS** — approve PR #7249 at head `97c598e9273737680fa5558c2b5efdd6edfb2ba9`. All three scope items are delivered correctly; placement policy is untouched; two informational nits only (no blockers).

**Findings**

1. **(a) Placement policy unchanged — verified byte-identical.** `decide_dispatch_placement`, `notebook_fallback_after_forward`, and `forward_dispatch` in `scripts/orchestration/job_host_exec.py` are byte-identical to `main` (function-level `git show main:... | diff` → `IDENTICAL` for all three). In `cmd_dispatch` the forward block's control flow is preserved: same `if not notebook_fallback_after_forward(forward_rc, error=forward_error):` gate, same `return 2` on config error, same `return int(forward_rc or 0)` otherwise. The only change inside the refusal branch is the added record write + `format_forward_config_refusal` message (`scripts/delegate.py:5294-5318`). No new fallback trigger; fail-closed on config errors is intact. `ForwardConfigError` subclasses `ValueError` (`job_host_exec.py:87`), so the pre-existing `except ValueError` catch at `job_host_exec.py:700` still binds; exit codes unchanged (2 config, 255 transport).

2. **(b) Clobber guard reused — verified.** `_record_forward_failure` (`scripts/delegate.py:5180`) is a thin wrapper delegating to `_record_worktree_prep_failure`, which retains the #7244 pre-write re-check: `existing = _read_state(state_path); if existing ... status in ("running", "spawning"): return False` (`scripts/delegate.py:5173-5175`). Covered by `test_record_forward_failure_refuses_clobber_running_record`, which asserts a live `running` record is not overwritten.

3. **(c) Message accuracy — verified against actual reads.** Every env var named in `format_forward_config_refusal` (`job_host_exec.py:90-102`) matches a constant at `job_host_exec.py:45-51` and a real read site: `LU_JOB_DISPATCH_HOST`/`ATLAS_RUNNER_HOST` (:107-108), `LU_JOB_REPO` (:116), `LU_TEACHER_DISPATCH_HOST` (:143), `LU_DISPATCH_SSH` opaque map (:139), `LU_TEACHER_REPO` (:152), `LU_ALLOW_NOTEBOOK_DISPATCH` (:290, value compared to `"1"` — message says `=1`, accurate).

4. **(d) Worker's "10 hunks, 0 deletions" self-audit is inaccurate as stated — substance verified independently.** Actual diff: 16 hunks (9 in `delegate.py`, 7 in `job_host_exec.py`), and it does contain deletions — `ValueError` raise lines replaced by `ForwardConfigError`, the old `print(f"❌ VPS forward failed: ...")` replaced, and the `early_prompt` block moved (deleted at old site, re-inserted earlier). I diffed every touched function against `main` individually: all deletions are in-place replacements; no logic removed. **Nit (non-blocking):** the `cmd_dispatch` docstring changed `` `<task_id>\\n<run_nonce>` `` to `` `<task_id>\n<run_nonce>` `` in a non-raw docstring, so `cmd_dispatch.__doc__` now contains a literal newline instead of the two characters `\n` — cosmetic only; the `--help` epilog still uses the escaped form.

5. **(e) Both mutation checks reproduced by me.**
   - Broke the message (`"deliberately do not fall back"` → `"fall back silently"`): `2 failed` — `tests/test_delegate.py::test_forward_config_refusal_names_both_recovery_paths` and `tests/orchestration/test_job_host_exec.py::test_format_forward_config_refusal_names_both_recovery_paths` (`AssertionError: assert 'deliberately do not fall back' in '❌ VPS forward configuration refusal for host-job: ...'`).
   - Skipped the record write (`False and _record_forward_failure(...)`): `FAILED tests/test_delegate.py::test_forward_config_refusal_writes_terminal_task_record` (state file never written).
   - Restored both files; `git status --short` empty, `git diff --quiet && git diff --cached --quiet` → `CLEAN`; re-ran the 7 refusal/record tests → `7 passed`.

**Full suite (my run, primary interpreter, worktree root):**
```
/Users/krisztiankoos/projects/learn-ukrainian/.venv/bin/pytest -q tests/test_delegate.py tests/test_delegate_readonly_guard.py tests/test_delegate_runtime_tmp_sweep.py tests/test_delegate_ownership.py tests/test_delegate_primary_integrity.py tests/test_delegate_check_budget.py tests/test_delegate_timeouts.py tests/test_subprocess_timeout_guard.py tests/orchestration/test_job_host_exec.py
======================== 453 passed in 86.22s (0:01:26) ========================
```
Matches the driver's 453 passed. No edits, commits, or pushes left behind; tree clean at head `97c598e9273737680fa5558c2b5efdd6edfb2ba9`.

_Review of record: kimi (Moonshot K3), cross-family vs author agy (Gemini); task review-7230, head 97c598e927, both mutation checks executed by the reviewer; posted by the infra driver._

--- a/tests/test_delegate.py
+++ b/tests/test_delegate.py
@@ -9488,19 +9488,20 @@ def test_forward_config_refusal_names_both_recovery_paths(tmp_tasks_dir, monkeyp
         "decide_dispatch_placement",
         lambda **_kwargs: ("vps", "available", "host-job"),
     )
-    args = delegate.build_parser().parse_args(
-        [
-            "dispatch",
-            "--agent",
-            "codex",
-            "--task-id",
-            "forward-refusal-7230",
-            "--initiator",
-            "codex",
-            "--prompt",
-            "test forward config refusal",
-        ]
-    )
+    raw_argv = [
+        "scripts/delegate.py",
+        "dispatch",
+        "--agent",
+        "codex",
+        "--task-id",
+        "forward-refusal-7230",
+        "--initiator",
+        "codex",
+        "--prompt",
+        "test forward config refusal",
+    ]
+    monkeypatch.setattr(sys, "argv", raw_argv)
+    args = delegate.build_parser().parse_args(raw_argv[1:])
 
     rc = delegate.cmd_dispatch(args)
     assert rc == 2
@@ -9530,19 +9531,20 @@ def test_forward_config_refusal_writes_terminal_task_record(tmp_tasks_dir, monke
         lambda **_kwargs: ("vps", "available", "host-job"),
     )
     task_id = "forward-record-7230"
-    args = delegate.build_parser().parse_args(
-        [
-            "dispatch",
-            "--agent",
-            "codex",
-            "--task-id",
-            task_id,
-            "--initiator",
-            "codex",
-            "--prompt",
-            "test record write on refusal",
-        ]
-    )
+    raw_argv = [
+        "scripts/delegate.py",
+        "dispatch",
+        "--agent",
+        "codex",
+        "--task-id",
+        task_id,
+        "--initiator",
+        "codex",
+        "--prompt",
+        "test record write on refusal",
+    ]
+    monkeypatch.setattr(sys, "argv", raw_argv)
+    args = delegate.build_parser().parse_args(raw_argv[1:])
 
     rc = delegate.cmd_dispatch(args)
     assert rc == 2

